algo/vector_msg[] vec